draft-ietf-gnap-core-protocol.md

```diff
@@ -1086,6 +1086,8 @@ object with the following fields.
 
 Additional fields are defined in the [Client Instance Fields Registry](#IANA-client-instance).
 
+Both the "display" and "class_id" are self-declarative and thus the AS SHOULD exercize caution in their interpretation, taking them as a hint but not as absolute truth. The "class_id" field can be used in a variety of ways to help the a variety of ways to help the AS make sense of the particular context in which the client is operating. In corporate environments for instance, different levels of trust might apply depending on security policies, and this field might help the AS adjust its own access decisions. It is also possible to configure a set of accepted values and rules during a pre-registration, so that expectations are defined prior using them during runtime requests. In other cases, the client runs with a specific AS in mind, so an hardcoded value would be acceptable (for instance, a set top box with a "class_id" claiming to be "FooBarTV version 4"). Or yet while the client instance may not have contacted the AS yet, the value of this "class_id" field may also be evaluated by the AS according to a broader context of dynamic use, alongside other related information available elsewhere (for instance, corresponding fields in AS make sense of the particular context in which the client is operating. In corporate environments different levels of trust might apply depending on security policies, and this field aims to help the AS adjust its own access decisions. It is possible to configure a set of values and rules during a pre-registration, and then later in runtime as a hint to the AS. In other cases, the client runs with a specific AS in mind, so an hardcoded value would acceptable (for instance, a set top box with a "class_id" claiming to be "FooBarTV version 4"). Or while the client instance may not have contacted the AS yet, the value of this "class_id" field may also be evaluated by the AS according to a broader context of dynamic use, alongside other related information available elsewhere (for instance, corresponding fields in a certificate). If the AS is not able to interpret the class_id field, it MAY return an "invalid_client" error or choose to return lesser levels of privileges.
```

Comment:
I think there are some copy-paste errors here, I tried to trim this down to a single paragraph without the additional sentences.

```suggestion
Both the `display` and `class_id` are self-declarative and thus the AS SHOULD exercise caution in their interpretation, taking them as a hint but not as absolute truth. The `class_id` field can be used in a variety of ways to help the a variety of ways to help the AS make sense of the particular context in which the client instance is operating. In corporate environments, for example, different levels of trust might apply depending on security policies. This field aims to help the AS adjust its own access decisions for different classes of client software. It is possible to configure a set of values and rules during a pre-registration, and then have the client instances provide them later in runtime as a hint to the AS. In other cases, the client runs with a specific AS in mind, so a single hardcoded value would acceptable (for instance, a set top box with a `class_id` claiming to be "FooBarTV version 4"). While the client instance may not have contacted the AS yet, the value of this `class_id` field can be evaluated by the AS according to a broader context of dynamic use, alongside other related information available elsewhere (for instance, corresponding fields in a certificate). If the AS is not able to interpret the class_id field, it SHOULD return an `invalid_client` error or choose to return lesser levels of privileges. See additional discussion of client instance impersonation in {{security-impersonation}}.
```